core/src/snapshot_packager_service.rs

```diff
@@ -64,19 +81,31 @@ impl SnapshotPackagerService {
                     let snapshot_slot = snapshot_package.slot;
                     let snapshot_hash = snapshot_package.hash;
 
+                    if exit_backpressure.is_some() {
+                        // With exit backpressure, we will delay flushing snapshot storages
+                        // until we receive a graceful exit request.
+                        // Save the snapshot storages here, so we can flush later (as needed).
+                        teardown_state = Some(TeardownState {
+                            snapshot_slot: snapshot_package.slot,
+                            snapshot_storages: snapshot_package.snapshot_storages.clone(),
```

Comment:
Can we change snapshot_package.snapshot_storages form Vec<_> to Arc<Vec<_>>? 

This may help to save allocation of a Vec and copying all the element of the Vec.